k9mail-library/src/test/java/com/fsck/k9/mail/internet/ListHeadersTest.java

```diff
@@ -0,0 +1,74 @@
+package com.fsck.k9.mail.internet;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE, sdk = 21)
+public class ListHeadersTest {
+    private String[] testEmails = new String [] {
+        "prettyandsimple@example.com",
+        "very.common@example.com",
+        "disposable.style.email.with+symbol@example.com",
+        "other.email-with-dash@example.com",
+        //TODO: Fix Address.parse()
+        /**
+            "\"much.more unusual\"@example.com",
+            "\"very.unusual.@.unusual.com\"@example.com",
+            //"very.(),:;<>[]\".VERY.\"very@\\ \"very\".unusual"@strange.example.com
+            "\"very.(),:;<>[]\\\".VERY.\\\"very@\\\\ \\\"very\\\".unusual\"@strange.example.com",
+            "admin@mailserver1",
+            "#!$%&'*+-/=?^_`{}|~@example.org",
+            "\"()<>[]:,;@\\\\\\\"!#$%&'*+-/=?^_`{}| ~.a\"@example.org",
+            "\" \"@example.org",
+            "example@localhost",
+            "example@s.solutions",
+            "user@com",
+            "user@localserver",
+            "user@[IPv6:2001:db8::1]"
+        **/
+    };
+
+    @Test
+    public void parsePostAddress_should_be_able_to_parse_mailto() {
+        for (String testEmail : testEmails) {
+            assertEquals(1,
+                    ListHeaders.parsePostAddress(
+                            "<mailto:" + testEmail + ">").length);
```

Comment:
Unfortunately, certain characters in `mailto:` URIs have to be encoded. It's not a simple matter of prepending `mailto:`.

See e.g. [EmailIntentBuilder.encodeRecipient()](https://github.com/cketti/EmailIntentBuilder/blob/2629ccf4cda320a45fbd546c89ac466b8452faf1/library/src/main/java/de/cketti/mailto/EmailIntentBuilder.java#L316-L321) and [the respective tests](https://github.com/cketti/EmailIntentBuilder/blob/2629ccf4cda320a45fbd546c89ac466b8452faf1/library/src/test/java/de/cketti/mailto/EmailIntentBuilderTest.java#L183-L202).